src/VCEL.Core/Expression/Impl/MaybeNotEqExpr.cs

```diff
@@ -35,7 +35,7 @@ public TMonad Evaluate(IContext<TMonad> context)
                 return Monad.Lift(false);
             }
 
-            return Monad.Lift(!Equals(l.Value, r.Value));
+            return Monad.Lift(!TypeOperation.EqualsChecked(l.Value, r.Value));
```

Comment:
Ideally we would use `ComparableExpr` comparable evaluation however that currently returns a Monad.Unit when one side of the expression is null. Need to consider merging these two ways of comparing types in the future.